zio/zngio/reader.go

```diff
@@ -126,29 +141,24 @@ again:
 		case zng.TypeDefAlias:
 			err = r.readTypeAlias()
 		case zng.CtrlEOS:
-			if r.uncompressed != nil {
-				return nil, nil, errors.New("zngio: CtrlEOS in compressed data")
-			}
 			r.reset()
 		case zng.CtrlCompressed:
-			if r.uncompressed != nil {
-				return nil, nil, errors.New("zngio: CtrlCompressed in compressed data")
+			if err := r.readCompressed(); err != nil {
+				return 0, nil, err
 			}
-			err = r.readCompressed()
+			return -zng.CtrlCompressed, r.uncompressed.Bytes(), nil
```

Comment:
This is being exposed like this in preparation for the boyer-moore work, right? (instead of just reading the compressed data & then falling out to the `goto again`, like the `CtrlEOS` handling)

Reply:
never mind, I read down to `zngScanner.Pull`